XmlFormatter/src/Windows/PluginManager.cs

```diff
@@ -0,0 +1,245 @@
+        private Panel currentSettingsPanel;
+
+        /// <summary>
+        /// Create a new instance of the plugin manager window
+        /// </summary>
+        /// <param name="pluginManager">The plugin manager to use</param>
+        /// <param name="settingsManager">The settings manager to use</param>
+        /// <param name="settingsFileName">The file to save the settings in</param>
+        public PluginManager(IPluginManager pluginManager, ISettingsManager settingsManager, string settingsFileName)
+        {
+            InitializeComponent();
+            this.pluginManager = pluginManager;
+            this.settingsManager = settingsManager;
+            this.settingFile = settingsFileName;
+
+            currentSettingsPanel = null;
+
+            TC_PluginData.Enabled = false;
+            L_Name.Tag = L_Name.Text;
+            L_Version.Tag = L_Version.Text;
+            L_Author.Tag = L_Author.Text;
+            TB_Description.ReadOnly = true;
+        }
+
+        /// <summary>
+        /// Load event of the plugin window
+        /// </summary>
+        /// <param name="sender">Sender of the event</param>
+        /// <param name="e">The event arguments</param>
+        private void PluginManager_Load(object sender, EventArgs e)
+        {
+            TreeNode root = new TreeNode("Plugins");
+            TreeNode formatter = new TreeNode("Formatter");
+            AddPluginsOfType<IFormatter>(formatter);
+            root.Nodes.Add(formatter);
+
+            TreeNode updater = new TreeNode("Updater");
+            AddPluginsOfType<IUpdateStrategy>(updater);
+            root.Nodes.Add(updater);
+
+            TV_Plugins.Nodes.Add(root);
+        }
+
+        /// <summary>
+        /// Add new plugins to the tree view of a given type
+        /// </summary>
+        /// <typeparam name="T">The plugin type</typeparam>
+        /// <param name="node">The root node to create the plugins in</param>
+        private void AddPluginsOfType<T>(TreeNode node) where T : IPluginOverhead
+        {
+            List<PluginMetaData> pluginMetas = pluginManager.ListPlugins<T>();
+            foreach (PluginMetaData metaData in pluginMetas)
+            {
+                TreeNode selectedPlugin = new TreeNode(metaData.Information.Name)
+                {
+                    Tag = metaData
+                };
+                node.Nodes.Add(selectedPlugin);
+            }
+        }
+
+        /// <summary>
+        /// After item was selected in the tree view
+        /// </summary>
+        /// <param name="sender">The sender of the event</param>
+        /// <param name="e">The arguments of the evend</param>
+        private void TV_Plugins_AfterSelect(object sender, TreeViewEventArgs e)
+        {
+            currentPlugin = null;
+            RemoveSettingsTab();
+            if (sender is TreeView treeView)
+            {
+                if (treeView.SelectedNode is TreeNode node && node.Tag is PluginMetaData metaData)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Merge this if statement with the enclosing one.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5528129)